### ISSUE TITLE
Characters not escaped in commands

### ISSUE THREAD

**Issue by Saviq** [canonical/multipass#1495]
```console
$ multipass mount . primary:~/path/with\$dollarsign
$ multipass exec primary -- ls path/with\$dollarsign
ls: cannot access 'path/with$dollarsign': No such file or directory
$ multipass exec primary -- ls path/with
Dokumenty   Muzyka   Obrazy   Pobrane   Publiczny   Pulpit   Szablony   Wideo
```
_Originally posted by @Saviq in https://github.com/canonical/multipass/pull/1439#pullrequestreview-395244787_

**Comment by seb128** [comment]
Bug #1840 has been flagged as duplicate, there is a similar issue with the transfer command and other special chars (~ for the target directory or dir/*), the title could be made less specific if that's the same issue

**Comment by luis4a0** [comment]
Right, changed the title here, thanks!

**Comment by letanloc1998** [comment]
I have a same problem

**Comment by WLBF** [comment]
My problem seems related to this issue. with version `1.8.0`. I can not transfer hidden file with dot in its name like `.vimrc`.
```bash
$ multipass  transfer config.yaml master-0:/home/ubuntu/config.yaml # this is ok
$ multipass  transfer .vimrc master-0:/home/ubuntu/.vimrc
transfer failed: [sftp push] error opening file for reading: Permission denied
```

**Comment by Saviq** [comment]
@WLBF it's unrelated actually. The snap confinement prevents us from reading hidden files for security reasons:

https://snapcraft.io/docs/home-interface

**Comment by ricab** [comment]
This is still the case. I suppose we should just follow whatever `scp` does in here.

**Comment by ricab** [comment]
Bumped priority because this can have unexpected consequences:

```
$ multipass mount adir/ prodigious-mamba:with\$USER
mount failed: The following errors occurred:
error mounting "/home/ubuntu/with$USER": Process returned exit code: 1: sshfs: bad mount point `/home/ubuntu/withubuntu': No such file or directory

$ multipass exec -n prodigious-mamba -- ls
snap  withroot
```

**Comment by letanloc1998** [comment]
Are people still using Multipass? I switched to WSL and Docker because Multipass had too many issues that I could never fully resolve.

**Comment by townsend2010** [comment]
@letanloc1998 Kudos to you for an unhelpful comment for this particular issue! 🙄

**Comment by letanloc1998** [comment]
> [@letanloc1998](https://github.com/letanloc1998) Kudos to you for an unhelpful comment for this particular issue! 🙄

5 years but still not fixed

<img width="1650" height="297" alt="Image" src="https://github.com/user-attachments/assets/87a796e3-1256-4194-9577-9da959eb01f3" />

**Comment by ricab** [comment]
We're not idle and have been improving Multipass in many other ways. But we don't have infinite capacity and we need to choose some things over others. Whatever priorities we choose, they can never align with every user's needs or desires. Hopefully they align with some. 

In your case @letanloc1998, hopefully there is nothing you're entitled to that you aren't getting?

This is open source, so feel free to contribute back.

**Comment by townsend2010** [comment]
> 5 years but still not fixed

Yes, and that was pointed out by @ricab just previous to your unhelpful comment.  If you are no longer using Multipass, why are you even here making unhelpful comments?  It's a small team with big ambitions.  Please add something positive or just let it go.

**Comment by Jotanune** [comment]
Hi, I'd love to take a stab at this escaping bug. Let me know if I have the green light to work on it and submit a PR!

**Comment by tobe2098** [comment]
Go ahead!
